src/audio/module_adapter/module/generic.c

```diff
@@ -282,6 +282,20 @@ int module_process_legacy(struct processing_module *mod,
 	return 0;
 }
 
+bool module_is_ready_to_process(struct processing_module *mod,
+				struct sof_source __sparse_cache **sources,
+				int num_of_sources,
+				struct sof_sink __sparse_cache **sinks,
+				int num_of_sinks)
+{
+	struct module_data *md = &mod->priv;
+
+	if (!md->ops->is_ready_to_process)
+		return false;
```

Comment:
this should be `true` IMHO - if no method is implemented, by default always run. Returning `false` here effectively makes this method compulsory, because if it isn't implemented, it will never run

Reply:
and that is the intention
for LLs it does not matter, LLs must be ready at every cycle. For DP modules this must be implemented or the module will start every LL cycle, sometimes pointless. 
So I prefer "false" - it will force to implement the method. Or a runtime assert != NULL

Reply:
if it is compulsory, please check once at initialisation. And only for DP modules. And we have to make sure not to call this for LL modules, or replace this check with an `if (LL) return true;`